ethportal-api/src/types/portal_wire.rs

```diff
@@ -550,7 +550,14 @@ pub struct PopulatedOfferWithResult {
     /// The offered content key & value
     pub content_item: (RawContentKey, Vec<u8>),
     /// The channel to send the result of the offer to
-    pub result_tx: tokio::sync::mpsc::UnboundedSender<bool>,
+    pub result_tx: tokio::sync::mpsc::UnboundedSender<OfferTrace>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum OfferTrace {
+    Success(BitList<typenum::U8>),
+    Rejected,
+    Failed,
```

Comment:
"Rejected" sounds to me a bit like the peer thinks we offered something in an invalid way. To me "Declined" sounds more like they just don't want the data we offered.

Also, a quick docstring would be nice. A draft for you:
```suggestion
    /// Peer is not interested in any of the offered data
    Declined,
    /// This offer failed, perhaps locally or from a timeout or transfer failure
    Failed,
```

Reply:
Nice, gonna start applying this mindset to the women who don't want to go out on a date with me...